### PR TITLE
Ensure labels are preserved

### PR DESCRIPTION
**Pull request by aaronlevy**
This is a copy of https://github.com/kubernetes-incubator/bootkube/pull/387 with the conflicts resolved and bumped for v1.6 object types

fixes https://github.com/kubernetes-incubator/bootkube/issues/378

### CODE REVIEW

**Comment by aaronlevy** [comment]
rktbot run checkpointer tests

**Comment by aaronlevy** [comment]
rktbot run checkpointer tests

**Comment by pbx0** [comment]
restarting your test because tests were not running in parallel (that test job didn't get updated post prefetching containers PR)

**Comment by pbx0** [comment]
rktbot run checkpointer tests

**Comment by aaronlevy** [comment]
Soemthing that I'm seeing is that the pod-checkpointer (parent) and the checkpoint (static pod) just go into a loop where it starts the checkpoint, then stop the parent, then keeps repeating that process.

I don't immediately see a reason the change here would have any affect. The output of one of the pod-checkpointers showing that it thinks it should destroy itself:

```
I0418 00:51:39.165537       1 main.go:106] Trying to acquire the flock at "/var/run/lock/pod-checkpointer.lock"
I0418 00:51:39.881707       1 main.go:112] Starting checkpointer for node: 172.17.4.101
I0418 00:51:42.942464       1 main.go:388] Checkpoint manifest for kube-system/kube-apiserver-txjkq already exists. Skipping
I0418 00:51:42.945728       1 main.go:388] Checkpoint manifest for kube-system/kenc-sq186 already exists. Skipping
I0418 00:51:42.948393       1 main.go:388] Checkpoint manifest for kube-system/kube-etcd-0000 already exists. Skipping
I0418 00:51:42.954214       1 main.go:232] API GC: skipping inactive checkpoint kube-system/kube-apiserver-txjkq
I0418 00:51:42.954400       1 main.go:232] API GC: skipping inactive checkpoint kube-system/kube-etcd-0000
I0418 00:51:42.954429       1 main.go:241] API GC: should remove inactive checkpoint kube-system/pod-checkpointer-65lts
I0418 00:51:42.954446       1 main.go:257] API GC: should remove active checkpoint kube-system/pod-checkpointer-65lts
I0418 00:51:42.954466       1 main.go:610] Removing checkpoint of: kube-system/kube-etcd-0000
I0418 00:51:42.954640       1 main.go:610] Removing checkpoint of: kube-system/kenc-sq186
I0418 00:51:42.954763       1 main.go:610] Removing checkpoint of: kube-system/kube-apiserver-txjkq
I0418 00:51:42.954952       1 main.go:610] Removing checkpoint of: kube-system/pod-checkpointer-65lts
```

/cc @yifan-gu

**Comment by aaronlevy** [comment]
Actually it looks like it is being re-adopted by the controller even if the ownerRef is cleared -- so we may still need a new label to force it to be different enough that it's not adopted? Or maybe a placeholder ownerref to keep it from be re-adopted?

File: cmd/checkpoint/main_test.go [hunk 1]
```diff
@@ -355,6 +357,28 @@ func TestSanitizeCheckpointPod(t *testing.T) {
 				},
 			},
 		},
+			{
+				desc: "Labels are preserved and OwnerReferences are deleted",
```
Comment (yifan-gu) [inline cmd/checkpoint/main_test.go]: Nvm, this is commented out, guess you are doing some experiments.

**Comment by aaronlevy** [comment]
rktbot run checkpointer tests

**Comment by aaronlevy** [comment]
rktbot run checkpointer tests

**Comment by yifan-gu** [comment]
From the test log:
```
+ sudo rkt fetch quay.io/coreos/pod-checkpointer-dev: --trust-keys-from-https
```
This doesn't look right @pbx0

**Comment by pbx0** [comment]
@yifan-gu eek, yes something broke in the jenkins job it appears. That just prefetches the newly built container from quay once before starting tests. Not strictly necessary, but unsure why its not picking up the git hash all of a sudden.

**Comment by pbx0** [comment]
rktbot run checkpointer tests

**Pull request merged**